tests/test_settings.py

```diff
@@ -3757,6 +3756,9 @@ class CfgWithSubCommand(BaseSettings):
         (Annotated[SimpleSettings, 'annotation'], 'JSON'),
         (DirectoryPath, 'Path'),
         (FruitsEnum, '{pear,kiwi,lime}'),
+        (time.time_ns, 'time_ns'),
+        (foobar, 'foobar'),
+        (CliDummyParser.add_argument, 'CliDummyParser.add_argument'),
```

Comment:
Also, @hramezani I updated representation for non-local functions to use `__qualname__`, which seems more informative for class methods etc. compared to `__name__`. e.g., `CliDummyParser.add_argument` compared to `add_argument`. 

Let me know if this is ok.